### DISCUSSION TITLE
Source Manual Switch like vMix Layers

### DISCUSSION THREAD

**Discussion by DarkR** [obsproject/obs-studio#11540] [Development Questions] [answered]
I think it would be great to have an opportunity to switch sources by right click menu.
For example vMix has layers which gives you an option to change any source via layer list on the fly keeping the transformation data.
It would be pretty easy to change the source in OBS keeping its transformation data.

For now to do that we need more steps: 

1. create a new/existing source
2. copy a transformation data from an old source
3. paste it to the new one
4. delete the old source

That would speed up everything. For example we need to change 1 Cam to 2 Cam in split overlay window.

**Comment by Warchamp7** [comment]
You can accomplish something to this effect currently by using a Scene as a Source

1. Create a scene ex. `Split Overlay Left Camera`
2. Add your different cameras to that scene
3. Add `Split Overlay Left Camera` as a Source in your main composition scene

**Reply by DarkR** [reply]
Thanks for that suggestion. We already have been using this approach with Scenes included in Scenes. But sometime we need to make fast Duplicate of main Scene and change one childe scene to another (i.e. switch Cam 1 Scene to Cam 2).

**Comment by Fenrirthviti** [comment, marked as answer]
This sounds more likely to be a feature request, and we are not currently accepting feature requests on GitHub Discussions. Make sure you read any [pinned posts](https://github.com/obsproject/obs-studio/discussions/5206) on discussion boards before opening a new discussion. Please use our [forums](https://obsproject.com/forum/) or [Discord](https://obsproject.com/discord/) for further assistance.

Thank you!
